### PR TITLE
Feature - unshorten URL

### PR DESCRIPTION
**Pull request by jacc**
New feature: unshorten URLs using spoopy.link API

### CODE REVIEW

**Comment by jacc** [comment]
Apparently I have my old nickname commits in there? so thats cool

File: src/discord/commands/utilities/unshorten.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+        if(!link) {
+          throw new CommandError({
+              message: "Please specify a link."
+          });
+        }
+
+        const spoop = await checkSpoopy(link)
+        const embed = new RichEmbed()
+        .setTitle(`${spoop.safe === false ? "Unsafe link detected" : "Safe link detected"}`);
```
Comment (olvrb) [inline src/discord/commands/utilities/unshorten.ts]: Just do `spoop.safe ? [...]`

File: src/discord/commands/utilities/unshorten.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        .setColor(`${spoop.safe === false ? 0xFF0000 : 0x00ee00}`);
```
Comment (olvrb) [inline src/discord/commands/utilities/unshorten.ts]: > Just do `spoop.safe ? [...]`

same here

File: src/discord/commands/utilities/unshorten.ts (hunk 1 above)
Comment (jacc) [reply src/discord/commands/utilities/unshorten.ts]: I have that locally, haven't pushed changes

File: package.json [hunk 3]
```diff
@@ -15,7 +15,9 @@
   "dependencies": {
     "dd-botkit": "^1.0.9-1",
     "pg": "^7.4.3",
-    "typeorm": "^0.2.7"
+    "spoopylink": "^1.0.3",
```
Comment (nullpixel) [inline package.json]: Please remove this unused dependancy.

File: src/discord/commands/utilities/unshorten.ts [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    },
+    handler: async (message, next) => {
+        const [link] = message.args;
+
+        if(!link) {
+          throw new CommandError({
+              message: "Please specify a link."
+          });
+        }
+        var url = await uu.expand(link)
```
Comment (nullpixel) [inline src/discord/commands/utilities/unshorten.ts]: Use `const` here

File: src/discord/commands/utilities/unshorten.ts [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+        const embed = new RichEmbed();
+        embed.setDescription(`${url}`)
```
Comment (nullpixel) [inline src/discord/commands/utilities/unshorten.ts:33]: Perhaps don't use an embed here

File: src/discord/commands/utilities/unshorten.ts [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+        await message.channel.send(embed);
```
Comment (nullpixel) [inline src/discord/commands/utilities/unshorten.ts]: use `message.reply`

File: src/discord/commands/utilities/unshorten.ts [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+                    {
+                        name: "link",
+                        type: "string",
+                        required: true
+                    }
+                ]
+            )
+        ]
+    },
+    handler: async (message, next) => {
```
Comment (nullpixel) [inline src/discord/commands/utilities/unshorten.ts]: generally, the first arg should be named `msg` for consistency

**Review by nullpixel — CHANGES_REQUESTED** [review]

**Review by jacc — COMMENTED** [review]
cc @nullpixel, all changes that have been requested have been made

File: package.json [hunk 8]
```diff
@@ -19,7 +19,7 @@
     "common-tags": "^1.8.0",
     "dd-botkit": "^1.0.9-1",
     "pg": "^7.4.3",
-    "typeorm": "^0.2.7"
```
Comment (olvrb) [inline package.json:22]: Why was typeorm removed?

File: package.json [hunk 9]
```diff
@@ -19,7 +19,7 @@
...
+    "url-unshort": "^5.0.0"
```
Comment (olvrb) [inline package.json:22]: Still shouldn't use other dependencies when there are better ways to unshorten links.

**Review by olvrb — CHANGES_REQUESTED** [review]

File: src/discord/commands/utilities/unshorten.ts [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+                        required: true
+                    }
+                ]
+            )
+        ]
+    },
+    handler: async (msg, next) => {
+        const [link] = msg.args;
+
+        if(!link) {
```
Comment (olvrb) [inline src/discord/commands/utilities/unshorten.ts:26]: Link will always be defined, as it's a required argument. Also, `CommandError` is never actually imported.

**Review by olvrb — CHANGES_REQUESTED** [review]

**Pull request closed without merging**